jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -661,146 +609,116 @@ public boolean exists(String path)
      */
     public void scan()
     {
-        try (AutoLock l = _lock.lock())
-        {
-            reportScanStart(++_scanCount);
-            scanFiles();
-            reportDifferences(_currentScan, _prevScan);
-            _prevScan.clear();
-            _prevScan.putAll(_currentScan);
-            reportScanEnd(_scanCount);
-
-            for (Listener listener : _listeners)
-            {
-                try
-                {
-                    if (listener instanceof ScanListener)
-                        ((ScanListener)listener).scan();
-                }
-                catch (Throwable e)
-                {
-                    LOG.warn("Unable to scan", e);
-                }
-            }
-        }
+        int cycle = _scanCount.incrementAndGet();
+        reportScanStart(cycle);
+        Map<String, MetaData> currentScan = scanFiles();
+        reportDifferences(currentScan, _prevScan == null ? Collections.emptyMap() : Collections.unmodifiableMap(_prevScan));
+        _prevScan = currentScan;
+        reportScanEnd(cycle);
     }
 
     /**
      * Scan all of the given paths.
      */
-    public void scanFiles()
+    public Map<String, MetaData> scanFiles()
```

Comment:
Make it `private` because `MetaData` is package private.
Maybe make `MetaData` a `private` class too?

Reply:
Done.